mocks/timestamp.js

```diff
@@ -55,8 +55,42 @@ class Timestamp {
   }
 }
 
+//
+// Search data for possible timestamps and convert to class.
+function convertTimestamps(data, path = []) {
+  if (!data) {
+    return data;
+  }
+  // we need to avoid self-referencing DB's (can happen on db.get)
+  // Check we have not looped.  If we have, backout
+  if (path.includes(data)) {
+    return;
+  }
+
+  // Check if this object is or contains a timestamp
+  if (typeof data === 'object') {
+    const keys = Object.keys(data);
+    // if it is a timestamp, convert to the appropriate class
+    if (keys.find(k => k === 'seconds') && keys.find(k => k === 'nanoseconds')) {
+      return new Timestamp(data.seconds, data.nanoseconds);
```

Comment:
while changing the type of the data from object => Timestamp might sound risky, it only adds functionality (no data is lost)